fontspec-code-internal.dtx

```diff
@@ -1052,6 +1052,30 @@
 %    \end{macrocode}
 % \end{macro}
 %
+% \begin{macro}{\@@_get_variations:}
+% \cmd{\@@_get_variations:} builds the feature string representing the
+% current variation instance and/or axis settings.
+%    \begin{macrocode}
+\cs_new:Nn \@@_format_axis:nn
+  {
+    #1 = #2 ,
+  }
+\cs_new:Nn \@@_get_variations:
+  {
+    \tl_if_empty:NF \g_@@_instance_tl
+      {
+        instance = { \g_@@_instance_tl };
+      }
+      \prop_if_empty:NF \g_@@_rawvariations_prop
+      {
+        axis = {
+          \prop_map_function:NN \g_@@_rawvariations_prop \@@_format_axis:nn
+        };
+      }
+  }
+%    \end{macrocode}
+% \end{macro}
+%
```

Comment:
These are not made LuaTeX specific since I could imagine a similar interface applying to XeTeX too if they ever get Variable font support.